Documentation/src/case-studies/calendar.md

```diff
@@ -0,0 +1,350 @@
+  
+private static string FormatDay(DateOnly day)
+  => $"{day.Day,WidthOfDay}";
+```
+
+We can ignore the full weeks, because they are already 21 characters long. How do we distinguish the beginning of the month from the end? The week at the end of the month must start with the first day of the week. So wee pad accordingly from left or the right.
+  
+```cs
+private static string PadWeek(string formattedWeek, IGrouping<int, DateOnly> week)
+  => StartsOnFirstDayOfWeek(week)
+    ? $"{formattedWeek,-WidthOfAWeek}"
+    : $"{formattedWeek,WidthOfAWeek}";
+```
+
+Now it just boils down to, what is the start of the week? It might be a surprise, that this is actually the most difficult part of this program, because the `DayOfWeek` enum defines Sunday as the first day of the week, which is not true for the largest part of the world, including all of europe. But `CultureInfo` has our back, because it tells us in the `DateTimeFormat`, which Day of the Week is the first Day. However even that needs some calculation with `% DaysInAWeek`.
+
+The function `NthDayOfWeek` gives us a 0 based index beginning with the start of the week. So we can simply check with `is FirstDayOfTheWeek` that we are indeed on the first day of the week. And this works independent of the given culture. Sweet.
+
+```cs
+private static bool StartsOnFirstDayOfWeek(IGrouping<int, DateOnly> week)
+    => NthDayOfWeek(week.First().DayOfWeek) is FirstDayOfTheWeek;
+
+private static int NthDayOfWeek(DayOfWeek dayOfWeek)
+    => (dayOfWeek + DaysInAWeek - CultureInfo.CurrentCulture.DateTimeFormat.FirstDayOfWeek) % DaysInAWeek;
+```
+
+Now we have an `IEnumerable<IEnumerable<string>>`, where the inner one is still representing a single month, but each line is already a completeley formatted week.
+
+## Layouting the months together.
+
+At this point we could print a list of months, we just would need to join all the lines together and it would look like this. (shortend to 2 months)
+
+```
+        January      
+        1  2  3  4  5
+  6  7  8  9 10 11 12
+ 13 14 15 16 17 18 19
+ 20 21 22 23 24 25 26
+ 27 28 29 30 31      
+
+       February      
+                 1  2
+  3  4  5  6  7  8  9
+ 10 11 12 13 14 15 16
+ 17 18 19 20 21 22 23
+ 24 25 26 27 28 29   
+
+```
+
+The next step looks simple: we would need to take line 1, 8, 15 and create the first line of the final output:
+
+```
+       January              February                March        
+```
+
+And then lines 2, 9 and 16 would be combined to:
+```
+        1  2  3  4  5                 1  2                    1
+```
+
+To do this lazily we use `Chunk`.
+
+`Chunk` is supported with .NET 6, before that Funcky has nearly identical Replacement. (The return type is slightly different)
+
+Chunk is grouping a sequnce into multiple sequnces of the same length. In our case we want to make the sequence of 12 months in to 4 sequences of length 3.
+
+```cs
+const HorizontalMonths = 3;
```

Comment:
MonthsPerLine? MonthsPerRow?

Reply:
Fixed in 239b3a73af300f846afba75f4a992650b6fcb41b